dipy/workflows/reconst.py

```diff
@@ -21,6 +21,184 @@
+        data_bvals : string
+            Path to the bval files.
+        b0_threshold : float, optional
+            Threshold used to find b=0 directions (default 0.0)
+        small_delta : float
+            Small delta value used in generation of gradient table of provided
+            bval and bvec. (default: 0.0129)
+        big_delta : float
+            Big delta value used in generation of gradient table of provided
+            bval and bvec. (default: 0.0218)
+        laplacian : boolean
+            Boolean value to indicate Laplacian model type
+            (default: True)
+        positivity : boolean
+            Boolean value to indicate Positivity model type
+            If both positivity and laplacian are True, then both are applied.
+            (default: True)
+        save_metrics : list of strings
+            List of metrics to save.
+            Possible values: rtop, laplacian_signal, msd, qiv, rtap, rtpp,
+            ng, perng, parng
+            (default: [] (all))
+        out_dir : string, optional
+            Output directory (default: input file directory)
+        out_rtop : string, optional
+            Name of the rtop to be saved
+        out_lapnorm : string, optional
+            Name of the norm of laplacian signal to be saved
+        out_msd : string, optional
+            Name of the msd to be saved
+        out_qiv : string, optional
+            Name of the qiv to be saved
+        out_rtap : string, optional
+            Name of the rtap to be saved
+        out_rtpp : string, optional
+            Name of the rtpp to be saved
+        out_ng : string, optional
+            Name of the Non-Gaussianity to be saved
+        out_perng :  string, optional
+            Name of the Non-Gaussianity perpendicular to be saved
+        out_parng : string, optional
+            Name of the Non-Gaussianity parallel to be saved
+        laplacian_weighting :
+            Weighting value used in fitting the MAPMRI model in the laplacian
+            and both model types. (default: 0.05)
+        """
+        io_it = self.get_io_iterator()
+        for dwi, bval, bvec, out_rtop, out_lapnorm, \
+            out_msd, out_qiv, out_rtap, out_rtpp, out_ng, \
+            out_perng, out_parng in io_it:
+
+            logging.info('Computing MAPMRI metrics for {0}'.format(dwi))
+            img = nib.load(dwi)
+            data = img.get_data()
+            affine = img.affine
+            bvals, bvecs = read_bvals_bvecs(bval, bvec)
+
+            gtab = gradient_table(bvals=bvals, bvecs=bvecs,
+                                  small_delta=small_delta,
+                                  big_delta=big_delta,
+                                  b0_threshold=b0_threshold)
+
+            if not save_metrics:
+                save_metrics = ['rtop', 'laplacian_signal', 'msd',
+                                'qiv', 'rtap', 'rtpp',
+                                'ng', 'perng', 'parng']
+
+            radial_order = 6
```

Comment:
Do you know if this static variable works for all dataset? I think `radial_order` might be an input parameter? what do you think?

Reply:
@arokem What do you think, should this be an input parameter?

Not sure what radial order is.

Reply:
radial_order defines how many basis functions mapmri uses to approximate the data. radial_order=0 means it only uses the first, which is just an anisotropic gaussian. radial_order=6 makes it use 50 basis functions.

Changing radial_order therefore directly influences the estimates q-space indices. You can default it to e.g. 6, but you definitely shouldn't hardcore it (maybe it makes sense to set it to 4 for very noisy or small data sets for example)